qa/remote-clusters/src/test/java/org/opensearch/cluster/remote/test/RemoteClustersIT.java

```diff
@@ -123,10 +125,11 @@ public void testHAProxyModeConnectionWorks() throws IOException {
 
         RemoteConnectionInfo rci = cluster1Client().cluster().remoteInfo(new RemoteInfoRequest(), RequestOptions.DEFAULT).getInfos().get(0);
         logger.info("Connection info: {}", rci);
-        if (!rci.isConnected()) {    
+        if (!rci.isConnected()) {
             logger.info("Cluster health: {}", cluster1Client().cluster().health(new ClusterHealthRequest(), RequestOptions.DEFAULT));
         }
-        assertTrue(rci.isConnected());
+        assertBusy(() -> assertTrue(rci.isConnected()), 10, TimeUnit.SECONDS);
```

Comment:
The failure reported on this test isn't associated with the connection not being set, instead its `java.lang.AssertionError: Request breaker not reset to 0 on node: node_s0
Expected: <0L>
     but: was <6553888L>`
     
This change doesn't look like it addressing a leaking resource, can you help me understand how they are related?

Reply:
Thanks for the review @peternied .The error reported is different than what you've pasted. Its an assertion failure on `rci.isConnected()`. I've added the investigation details in the linked issue.

Reply:
(!) My apologizes, I must have been drilling in on the wrong issue.